google/ggrc-core
# [WIP] GGRC-8677 Implement Post-Put hook for valid roleable models

# Issue description

It was possible to create any object with mandatory roles via API Post call without these roles. Now this behavior is prohibited.

# Steps to test the changes

Create new Assessment with all mandatory roles, copy call data as fetch and delete access_control_list from it.
It should look like these:

fetch("http://localhost:8080/api/assessments", {"credentials":"include","headers":{"accept":"*/*","accept-language":"en-US,en;q=0.9","cache-control":"max-age=0","content-type":"application/json","if-match":"undefined","if-unmodified-since":"undefined","sec-fetch-dest":"empty","sec-fetch-mode":"cors","sec-fetch-site":"same-origin","x-requested-by":"GGRC","x-requested-with":"XMLHttpRequest","x-usertimezoneoffset":"180"},"referrer":"http://localhost:8080/audits/1","referrerPolicy":"no-referrer-when-downgrade","body":"[{\"assessment\":{\"test_plan_procedure\":true,\"assessment_type\":\"Control\",\"status\":\"Not Started\",\"sox_302_enabled\":false,\"send_by_default\":true,\"recipients\":\"Assignees,Creators,Verifiers\",\"title\":\"newnew\",\"audit\":{\"id\":1,\"type\":\"Audit\",\"href\":\"/api/audits/1\"},\"issue_tracker\":{},\"custom_attribute_definitions\":[{\"placeholder\":null,\"previous_id\":null,\"external_name\":null,\"definition_id\":null,\"multi_choice_options\":\"Documentation,Interview\",\"modified_by\":null,\"definition_type\":\"assessment\",\"default_value\":\"\",\"title\":\"Type\",\"selfLink\":\"/api/custom_attribute_definitions/1\",\"updated_at\":\"2018-08-24T09:33:15\",\"id\":1,\"helptext\":\"Assessment Category\",\"created_at\":\"2018-08-24T09:33:15\",\"type\":\"CustomAttributeDefinition\",\"attribute_type\":\"Dropdown\",\"context\":null,\"mandatory\":false,\"multi_choice_mandatory\":null}],\"custom_attribute_values\":[{\"attribute_objects\":null,\"attribute_value\":\"\",\"custom_attribute_id\":1}],\"errors\":{},\"context\":{\"href\":\"/api/contexts/5\",\"context_id\":null,\"type\":\"Context\",\"id\":5},\"design\":\"\",\"operationally\":\"\",\"start_date\":null}}]","method":"POST","mode":"cors"}); ;

Before changes - Assessment with empty Creators and Assignees would be created.
After changes - you'll get 400 Bad request.

# Solution description

Create new hook that checks the presence of users with all mandatory roles in objects' access_control_list. If any mandatory role is missing, you'll get 400 Bad request.

# Sanity checklist




# PR Review checklist
